core/DataAccess/Model.php

```diff
@@ -319,14 +319,30 @@ public function getSitesWithInvalidatedArchive($numericTable)
      * @param string $oldestToKeep Datetime string
      * @return array of IDs
      */
-    public function getArchiveIdsForDeletedSites($archiveTableName, $oldestToKeep)
+    public function getArchiveIdsForDeletedSites($archiveTableName)
     {
-        $sql = "SELECT DISTINCT idarchive FROM " . $archiveTableName . " a "
-            . " LEFT JOIN " . Common::prefixTable('site') . " s USING (idsite)"
-            . " WHERE s.idsite IS NULL"
-            . " AND ts_archived < ?";
+        $sql = "SELECT DISTINCT idsite FROM " . $archiveTableName;
+        $rows = Db::getReader()->fetchAll($sql, array());
 
-        $rows = Db::fetchAll($sql, array($oldestToKeep));
+        if (empty($rows)) {
+            return array(); // nothing to delete
+        }
+
+        $idSitesUsed = array_column($rows, 'idsite');
+
+        $model = new \Piwik\Plugins\SitesManager\Model();
+        $idSitesExisting = $model->getSitesId();
+
+        $deletedSites = array_diff($idSitesUsed, $idSitesExisting);
+
+        if (empty($deletedSites)) {
+            return array();
+        }
+        $deletedSites = array_values($deletedSites);
+
+        $sql = "SELECT DISTINCT idarchive FROM " . $archiveTableName . " WHERE idsite IN (".Common::getSqlStringFieldsArray($deletedSites).")";
+
+        $rows = Db::getReader()->fetchAll($sql, $deletedSites);
```

Comment:
If there are hundreds or thousands of deleted sites, will this be a problem? (since we're using placeholders)

Reply:
I reckon it's unlikely this happens but will change it indeed 👍 doesn't hurt

Reply:
Will also apply this to the `deleteArchiveIds` method that is in the same model

Reply:
Done 👍